envi/archs/arm/emu.py

```diff
@@ -120,13 +131,27 @@ def c1101(flags):
         c1101,
         ]
 
-class ArmEmulator(ArmModule, ArmRegisterContext, envi.Emulator):
+top_bits_32 = [(e_bits.u_maxes[4] ^ ((e_bits.u_maxes[4]>>x))) for x in range(4*8)]
+
+LSB_FMT = [0, 'B', '<H', 0, '<I', 0, 0, 0, '<Q',]
+MSB_FMT = [0, 'B', '>H', 0, '>I', 0, 0, 0, '>Q',]
+LSB_FMT_SIGNED = [0, 'b', '<h', 0, '<i', 0, 0, 0, '<q',]
+MSB_FMT_SIGNED = [0, 'b', '>h', 0, '>i', 0, 0, 0, '>q',]
+
+class ArmEmulator(ArmRegisterContext, envi.Emulator):
 
     def __init__(self):
-        ArmModule.__init__(self)
+        # if-then placeholders
+        self.itva = None
+        self.itflags = None
+        self.itcount = None
+
+        # for memory exclusive access:
+        self.mem_access_lock = threading.Lock()
```

Comment:
Is there any way in our emulator to ever trip this? Because as fair as I know, we run single threaded (unless it's an interface detail for handling the CoProcEmulator).

Reply:
discussion about whether we should.  there are reasons to design the system for thread-safety... such as full emulation of a multi-core processor (which i'm actually planning for another architecture right now).